files/scripts/benchmark_job.sh

```diff
@@ -1,34 +1,45 @@
 #!/bin/bash
+set -x
 
-if [ "$#" -gt 13 ] || [ "$#" -lt 7 ]; then
+helpmsg="benchmark_job.sh -w [WORKSPACE] -n [node] -c [compiler] -H [JENKINS_HOME (scripts)] -b [BUILD_NUMBER] -g [benchmark_gitbranch] -m [benchmark_name] -f compiler_flags -l link_flags -o benchmark_options -d benchmark_build_deps -r benchmark_run_deps -v (for set -ex)"
+
+if [ "$#" -gt 25 ] || [ "$#" -lt 15 ]; then
```

Comment:
This is not good enough. The best (and easiest) way is to see if there are any left overs *after* getopt.
If there are compulsory options, add a check after that.

ex: if [ ! $A ] || [ ! $B ] ...; then echo "A, B, ... are required parameters"; fi